game/ui/base/vequipscreen.cpp

```diff
@@ -277,6 +278,20 @@ void VEquipScreen::eventOccurred(Event *e)
 		    std::dynamic_pointer_cast<VEquipment>(this->selected->getEquipmentAt(mouseSlotPos));
 		if (equipment)
 		{
+			// Check if a passenger module can be removed
+			if (this->highlightedVehicle->getPassengers() >
+			    (this->highlightedVehicle->getMaxPassengers() - 4))
```

Comment:
You can replace the hardcoded `4` with something like `equipment->type->passengers`